src/gt4py/next/program_processors/runners/dace_fieldview/gtir_sdfg.py

```diff
@@ -490,7 +501,10 @@ def visit_Lambda(
         ]
 
         # inherit symbols from parent scope but eventually override with local symbols
-        lambda_symbols = self.global_symbols | {
+        lambda_symbols = {
+            sym: self.global_symbols[sym]
+            for sym in symbol_ref_utils.collect_symbol_refs(node.expr, self.global_symbols.keys())
+        } | {
             pname: dace_gtir_utils.get_tuple_type(arg) if isinstance(arg, tuple) else arg.data_type
```

Comment:
Just a question.
As far as I can tell this is strictly speaking not necessary.
It just filters out the symbols that are not used.
Is this correct?